src/tools/rbd_mirror/Replayer.cc

```diff
@@ -399,7 +403,17 @@ bool Replayer::stop_image_replayer(unique_ptr<ImageReplayer<> > &image_replayer)
   }
 
   if (image_replayer->is_running()) {
-    image_replayer->stop();
+    LambdaContext *ctx = new LambdaContext(
+        [&image_replayer, this] (int r) {
+          if (m_image_deleter) {
+            m_image_deleter->schedule_image_delete(
+                          image_replayer->get_local_pool_id(),
+                          image_replayer->get_local_image_id(),
+                          image_replayer->get_global_image_id());
+          }
+        }
+    );
+    image_replayer->stop(ctx);
```

Comment:
I have a problem here. The information we keep for deleting an image is it's local_id and global_image_id.
If the user disables mirroring for the primary image and then re-enables it, the global_image_id will be different, so we will not be able check if there is a deletion scheduled. 
We could also keep the remote_id of the primary image, but it will also not work in the case where the user deletes and re-creates the primary image.

What will be the consequence of create a new ImageReplayer when there is a pending delete? Is because there still exists a local image with the same name? I that is the case, maybe what we should keep track in the DeleteInfo structure is the image name, which is common in the remote and local cluster.